### ISSUE TITLE
python-ldap wheel doesn't (didn't) build when using enhance-image.sh

### ISSUE THREAD

**Issue by dmckeone** [getsentry/self-hosted#3226]
### Self-Hosted Version

24.7.1

### CPU Architecture

x86_64

### Docker Version

20.10.11

### Docker Compose Version

2.24.6

### Steps to Reproduce

Using `sentry/enhance-image.sh` I was previously (several months ago) able to install `python-ldap` with the example [enhance-image.example.sh](https://github.com/getsentry/self-hosted/blob/24.7.1/sentry/enhance-image.example.sh) content.

With a recent upgrade to 24.7.1 I was unable to build the wheel due to some kind of gcc error when running `./install.sh`: `error: command 'gcc' failed: No such file or directory`

After a little searching, changing the content of `enhance-image.sh` from:

```bash
#!/bin/bash

# Enhance the base $SENTRY_IMAGE with additional dependencies, plugins - see https://github.com/getsentry/self-hosted#enhance-sentry-image
# For example:
apt-get update
apt-get install -y gcc libsasl2-dev python-dev libldap2-dev libssl-dev
pip install python-ldap
```

to

```bash
#!/bin/bash

# Enhance the base $SENTRY_IMAGE with additional dependencies, plugins - see https://github.com/getsentry/self-hosted#enhance-sentry-image
# For example:
apt-get update
apt-get install -y gcc libsasl2-dev python-dev-is-python-3 libldap2-dev libssl-dev
pip install python-ldap
```

solved the problem

TL;DR: change `python-dev` dependency to `python-dev-is-python-3` dependency, and things work.

Mentioning it here, just in case it helps others.

### Expected Result

`./install.sh` runs normally

### Actual Result

`./install.sh` output a wheel building error like: `error: command 'gcc' failed: No such file or directory`

**Comment by asottile-sentry** [comment]
I suspect that the `apt install` command didn't actually run -- the `python-dev` is a red herring (though that package doesn't exist any more) -- that would cause `gcc` to be missing

I'll update the sample and remove `python-dev` entirely since that's not needed any more
